src/developer-quickstart.md

```diff
@@ -0,0 +1,441 @@
+
+##### Install
+
+Inside `counter-dapp/frontend` run;
+
+```sh
+npm install fuels --save
+npm install fuelchain typechain-target-fuels --save-dev
+```
+
+##### Generating contract types
+
+To make it easier to interact with our contract we use `fuelchain` to interpret the output ABI JSON from our contract. This JSON was created on the moment we executed the `forc build` to compile our Sway Contract into binary. If you see the folder `counter_dapp/contract/out` you will be able to see the ABI JSON there. If you want to learn more, read the [ABI Specs here](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/abi.md).
+
+Inside `counter-dapp/frontend` run;
+
+```sh
+npx fuelchain --target=fuels --out-dir=./src/contracts ../fuel-counter/out/debug/*-abi.json
+```
+
+You should see something like this:
+
+```sh
+Successfully generated 4 typings!
+```
+
+Now you should be able to find a new folder `counter-dapp/frontend/src/contracts`. This folder was auto-generated by our `fuelchain` command, this files abstract the work we would need to do to create a contract instance, and generate a complete TypeScript interface to the Contract, making easy to develop.
+
+### Create A Wallet
+
+In the root of the frontend project create a file, createWallet.js
+
+```js
+const { Wallet } = require("fuels");
+
+const wallet = Wallet.generate();
+
+console.log("address", wallet.address.toString());
+console.log("private key", wallet.privateKey);
+```
+
+In a terminal, run the following command: `node createWallet.js`. You should see an output that looks like this:
+
+```console
+address fuel160ek8t7fzz89wzl595yz0rjrgj3xezjp6pujxzt2chn70jrdylus5apcuq
+private key 0x719fb4da652f2bd4ad25ce04f4c2e491926605b40e5475a80551be68d57e0fcb
+```
+
+Copy the private key, you will need this later to set it as a string value for a variable `WALLET_SECRET` in your `App.tsx` file. More on that below.
+
+First, take the address of your wallet and use it to get some coins from [the testnet faucet](https://faucet-beta-1.fuel.network/).
+
+Now you're ready to build and ship ⛽
+
+> Note: The team is working to simplify the process of creating a wallet, and eliminate the need to create a wallet twice. Keep an eye out for these updates.
+
+### Modify the App
+
+Inside the `frontend` folder let's add code that interacts with our contract.
+Read the comments to help you understand the App parts.
+
+Change the file `counter-dapp/frontend/src/App.tsx` to:
+
+```ts
+import React, { useEffect, useState } from "react";
+import { Wallet } from "fuels";
+import "./App.css";
+// Import the contract factory -- you can find the name in index.ts.
+// You can also do command + space and the compiler will suggest the correct name.
+import { CounterDappAbi__factory } from "./contracts";
+// The address of the contract deployed the Fuel testnet
+const CONTRACT_ID = "0xaf94c0a707756caae667ee43ca18bace441b25998c668010192444a19674dc4f";
```

Comment:
We should keep on the place of the private key and contract id a placeholder string to make sure users updates with their privateKey and contract id.

```ts
// The address of the contract deployed the Fuel testnet
const CONTRACT_ID = "<replace with your CONTRACT_ID here>";
//the private key from createWallet.js
const WALLET_SECRET = "<replace with your WALLET_SECRET here>";
```